flink-yarn/src/test/java/org/apache/flink/yarn/TestingContainerStatus.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.yarn;
+
+import org.apache.hadoop.yarn.api.records.ContainerId;
+import org.apache.hadoop.yarn.api.records.ContainerState;
+import org.apache.hadoop.yarn.api.records.ContainerStatus;
+
+/**
+ * A {@link ContainerStatus} implementation for testing.
+ */
+class TestingContainerStatus extends ContainerStatus {
+
+	private final ContainerId containerId;
+	private final ContainerState containerState;
+	private final String diagnostics;
+	private final int exitStatus;
+
+	TestingContainerStatus(
+		final ContainerId containerId,
+		final ContainerState containerState,
+		final String diagnostics,
+		final int exitStatus) {
+
+		this.containerId = containerId;
+		this.containerState = containerState;
+		this.diagnostics = diagnostics;
+		this.exitStatus = exitStatus;
+	}
+
+	@Override
+	public ContainerId getContainerId() {
+		return containerId;
+	}
+
+	@Override
+	public void setContainerId(ContainerId containerId) {
+
```

Comment:
Shouldn't we fail in case someone calls `setContainerId` here? Otherwise it might go unnoticed and result in some strange behaviour/failure.